### PR TITLE
Add support for Circle CI for Linuxbrew

### PR DESCRIPTION
**Pull request by sjackman**

### CODE REVIEW

**Comment by sjackman** [comment]
Travis: `Error: GitHub API Error: API rate limit exceeded`
https://travis-ci.org/Homebrew/homebrew-test-bot/jobs/179313656#L928

File: cmd/brew-test-bot.rb [hunk 1]
```diff
@@ -373,6 +373,10 @@ def diff_formulae(start_revision, end_revision, path, filter)
       elsif travis_pr
         @url = "https://github.com/#{ENV["TRAVIS_REPO_SLUG"]}/pull/#{ENV["TRAVIS_PULL_REQUEST"]}"
         @hash = nil
+      elsif ENV["CI_PULL_REQUEST"] && !ENV["CI_PULL_REQUEST"].empty?
+        # Circle CI build of a pull request.
```
Comment (MikeMcQuaid) [inline cmd/brew-test-bot.rb]: `# Use Circle CI pull-request variables for pull request jobs.` and put it above `elsif ENV["CI_PULL_REQUEST"]`. Just to check: is this actually empty but set, ever?

File: cmd/brew-test-bot.rb [hunk 2]
```diff
@@ -948,6 +952,7 @@ def test_ci_upload(tap)
       jenkins = ENV["JENKINS_HOME"]
       job = ENV["UPSTREAM_JOB_NAME"]
       id = ENV["UPSTREAM_BUILD_ID"]
+      return unless jenkins
       raise "Missing Jenkins variables!" if !jenkins || !job || !id
```
Comment (MikeMcQuaid) [inline cmd/brew-test-bot.rb]: Remove `!jenkins` here because it'll never be unset.

File: cmd/brew-test-bot.rb [hunk 3]
```diff
@@ -975,7 +980,7 @@ def test_ci_upload(tap)
     safe_system "git", "reset", "--hard", "origin/master"
     safe_system "brew", "update"
 
-    if (pr = ENV["UPSTREAM_PULL_REQUEST"])
+    if (pr = ENV["UPSTREAM_PULL_REQUEST"] || ENV["CIRCLE_PR_NUMBER"])
```
Comment (MikeMcQuaid) [inline cmd/brew-test-bot.rb]: Pull this into a separate variable and document that the variables are for Jenkins and Circle CI respectively.

File: cmd/brew-test-bot.rb [hunk 4]
```diff
@@ -989,7 +994,7 @@ def test_ci_upload(tap)
     remote = "git@github.com:BrewTestBot/homebrew-#{tap.repo}.git"
     git_tag = if pr
       "pr-#{pr}"
-    elsif (upstream_number = ENV["UPSTREAM_BUILD_NUMBER"])
+    elsif (upstream_number = ENV["UPSTREAM_BUILD_NUMBER"] || ENV["CIRCLE_BUILD_NUM"])
```
Comment (MikeMcQuaid) [inline cmd/brew-test-bot.rb]: Pull this into a separate variable and document that the variables are for Jenkins and Circle CI respectively.

File: cmd/brew-test-bot.rb (hunk 1 above)
Comment (sjackman) [inline cmd/brew-test-bot.rb]: As I recall, Circle sets it to the empty string for non-PR builds, that is branch builds.

**Comment by sjackman** [comment]
Thanks for your comments, Mike. I've addressed them in this updated PR.

**Comment by sjackman** [comment]
Thanks for the merge, Mike.

**Pull request merged**